addons/project/models/project.py

```diff
@@ -614,6 +660,23 @@ def _create_analytic_account(self):
             })
             project.write({'analytic_account_id': analytic_account.id})
 
+    def _check_project_sharing_access(self, access_mode='read'):
+        self.ensure_one()
+        if access_mode not in map(lambda access: access[0], PROJECT_SHARING_ACCESS_MODE):
+            raise ValidationError('The %s does not exists in project sharing accesses.' % access_mode)
+        if self.env.user._is_public():
+            return False
+        if self.env.user.has_group('base.group_portal'):
+            # check in project sharing access model of the project
+            project_sharing_access = self.project_sharing_access_ids.filtered(lambda access: access.user_id == self.env.user)
+            if not project_sharing_access:
+                return False
+            if access_mode == 'edit':
+                return project_sharing_access.access_mode == 'edit'
+            elif access_mode == 'comment':
+                return project_sharing_access.access_mode in ('comment', 'edit')
+        return True  # because he is a internal user or the portal user has readonly access mode
```

Comment:
better make that last check explicit, and not just rely on "generally expected situations" ;)

```suggestion
        return self.env.user.has_group('base.group_user')
```